examples/fidelity.py

```diff
@@ -0,0 +1,86 @@
+"""Simulates the fidelity estimation.
+
+Direct Fidelity Estimation from Few Pauli Measurements
+https://arxiv.org/abs/1104.4695
+
+Practical characterization of quantum devices without tomography
+https://arxiv.org/abs/1104.3835
+"""
+
+import cirq
+import heapq
+import itertools
+import math
+import numpy
+
+
+def build_circuit():
+  # Builds an arbitrary circuit to test. The circuit is non Clifford to show the
+  # use of simulators.
+  qubits = cirq.LineQubit.range(3)
+  circuit = cirq.Circuit(
+      cirq.Z(qubits[0])**0.25,  # T-Gate, non Clifford.
+      cirq.X(qubits[1])**0.123,
+      cirq.X(qubits[2])**0.456)
+  return circuit, qubits
+
+
+def simulate_trace(circuit, pauli_gates, qubits, noise):
```

Comment:
The flow of this example could be greatly assisted by following the nomenclature of one of the papers and including line comments describing which equation number each quantity corresponds to. For example: I think this function is calculating the "characteristic function" chi from https://arxiv.org/abs/1104.4695 where `circuit` specifies the construction of rho or sigma and `pauli_gates` corresponds to W_k

Reply:
Renamed some variables and added some comments. It may be that the two papers propose slightly different algo, so I am not sure which to follow... or I may just be plain confused.